internal/adapters/cloudformation/aws/elasticsearch/domain.go

```diff
@@ -26,18 +30,27 @@ func getDomains(ctx parser.FileContext) (domains []elasticsearch.Domain) {
 			AtRestEncryption: elasticsearch.AtRestEncryption{
 				Metadata: r.Metadata(),
 				Enabled:  defsecTypes.BoolDefault(false, r.Metadata()),
+				KmsKeyId: defsecTypes.String("", r.Metadata()),
 			},
 			Endpoint: elasticsearch.Endpoint{
 				Metadata:     r.Metadata(),
 				EnforceHTTPS: defsecTypes.BoolDefault(false, r.Metadata()),
 				TLSPolicy:    defsecTypes.StringDefault("Policy-Min-TLS-1-0-2019-07", r.Metadata()),
 			},
+			ServiceSoftwareOptions: elasticsearch.ServiceSoftwareOptions{
+				Metadata:        r.Metadata(),
+				CurrentVersion:  defsecTypes.String("", r.Metadata()),
+				NewVersion:      defsecTypes.String("", r.Metadata()),
+				UpdateStatus:    defsecTypes.String("", r.Metadata()),
+				UpdateAvailable: defsecTypes.Bool(false, r.Metadata()),
+			},
 		}
 
 		if prop := r.GetProperty("LogPublishingOptions"); prop.IsNotNil() {
 			domain.LogPublishing = elasticsearch.LogPublishing{
-				Metadata:     prop.Metadata(),
-				AuditEnabled: prop.GetBoolProperty("AUDIT_LOGS.Enabled", false),
+				Metadata:              prop.Metadata(),
+				AuditEnabled:          prop.GetBoolProperty("AUDIT_LOGS.Enabled", false),
```

Comment:
ditto here. is there a resource I can read which explains these fields?

Reply:
https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-elasticsearch-domain.html

Reply:
thanks makes sense